### PR TITLE
nixos/frr: refactor

### LINKED ISSUE #316163: nixos/frr: staticd config not working at first startup after boot
### Describe the bug
After applying https://github.com/NixOS/nixpkgs/pull/274425 which I remember running fine with frr-9 we now run into
https://github.com/FRRouting/frr/issues/15812

`static` configuration does not get recognized anymore, it has to be moved to `mgmt` to work again. The logic to enable mgmt when static is enabled is not the right solution anymore.

### Steps To Reproduce
take a config which worked before, like
```
services.frr.static = {
  enable = true;
  config = ''
    ip route 10.0.0.0/8 Null0
    ip route 172.16.0.0/12 Null0
    ip route 192.168.0.0/16 Null0
  '';
};

```

and get

```
mgmtd[852]: [SHWNK-NWT5S][EC 100663304] No such command on config line 3: hostname myhost
mgmtd[852]: [SHWNK-NWT5S][EC 100663304] No such command on config line 4: log syslog
mgmtd[852]: [SHWNK-NWT5S][EC 100663304] No such command on config line 5: service password-encryption
mgmtd[852]: [SHWNK-NWT5S][EC 100663304] No such command on config line 7: ip route 10.0.0.0/8 Null0
mgmtd[852]: [SHWNK-NWT5S][EC 100663304] No such command on config line 8: ip route 172.16.0.0/12 Null0
mgmtd[852]: [SHWNK-NWT5S][EC 100663304] No such command on config line 9: ip route 192.168.0.0/16 Null0
mgmtd[852]: [SHWNK-NWT5S][EC 100663304] No such command on config line 3: hostname myhost
mgmtd[852]: [SHWNK-NWT5S][EC 100663304] No such command on config line 4: log syslog
mgmtd[852]: [SHWNK-NWT5S][EC 100663304] No such command on config line 5: service password-encryption
```

and a empty staticd running config with frr-10, instead of the expected routes.

### Workaround
This can be worked around like

```
services.frr = {
    static.enable = true;
    mgmt = {
      # RFC1918 default discard
      config = ''
        ip route 10.0.0.0/8 Null0
        ip route 172.16.0.0/12 Null0
        ip route 192.168.0.0/16 Null0
      '';
    };
};
```

### Solution?
I think we have to change `nixos/modules/services/networking/frr.nix` to put "static" configuration into mgmt, to not break existing setups. Later we have to change to unified config at all. @mweinelt 

### Metadata
 - system: `"x86_64-linux"`
 - host os: `Linux 6.6.32, NixOS, 24.05 (Uakari), 24.05.20240530.0f1a94c`
 - multi-user?: `yes`
 - sandbox: `yes`
 - version: `nix-env (Nix) 2.18.2`
 - channels(fd): `""`
 - channels(root): `""`
 - nixpkgs: `/nix/store/v0sia9lh4vl46dsx8dcw1q5vpj1gzndm-source`

### PR DESCRIPTION
**Pull request by woffs**
## Description of changes

- use upstream service and scripts
- switch to integrated-vtysh-config, abandon per-daemon config
- use always daemon names in options (e.g. ospf -> ospfd)
- zebra, mgmtd and staticd are always enabled
- abandon vtyListenAddress, vtyListenPort options; use "options"/"extraOptions" instead
- extend test to test staticd
- pkgs.servers.frr: fix sbindir and remove FHS PATH

fixes: #316163

## Things done

- Built on platform(s)
  - [X] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [X] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [24.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2411.section.md) (or backporting [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) and [24.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [X] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

### CODE REVIEW

**Comment by woffs** [comment]
@GrahamcOfBorg test frr

**Comment by woffs** [comment]
@GrahamcOfBorg test frr

**Review by Anillc — APPROVED** [review]
Looks great

File: nixos/modules/services/networking/frr.nix
Comment (thillux) [inline nixos/modules/services/networking/frr.nix:318]: Please add a comment here, that this should be increased, in setups with many routing peers (e.g. BGP) and/or management connections.

File: nixos/modules/services/networking/frr.nix
Comment (thillux) [inline nixos/modules/services/networking/frr.nix:301]: NixOS does not have this service definition. Drop this line? I'll see that you kept it here for comparison with upstream systemd definition. But I would vote for dropping it.

File: nixos/modules/services/networking/frr.nix
Comment (thillux) [inline nixos/modules/services/networking/frr.nix:253]: systemd already limits this. Larger values than in systemd service definition will have no effect I guess. Upstream has this definition when systemd is not used I guess.

**Review by thillux — APPROVED** [review]
LGTM, some small comments added.

File: nixos/modules/services/networking/frr.nix
Comment (woffs) [reply nixos/modules/services/networking/frr.nix:318]: or even make a `services.frr` option out of this?

File: nixos/modules/services/networking/frr.nix
Comment (woffs) [reply nixos/modules/services/networking/frr.nix:253]: I think I'll strip the comments from the verbatim included (upstream) `/etc/frr/daemons` template. If anyone really needs `frr_profile` or other options, we can still extend this later anyway.

File: nixos/modules/services/networking/frr.nix
Comment (woffs) [reply nixos/modules/services/networking/frr.nix]: I have introduced `services.frr.openFilesLimit`, seemed to be the most clean way.

**Review by thillux — APPROVED** [review]

File: nixos/doc/manual/release-notes/rl-2205.section.md
Comment (JohnRTitor) [inline nixos/doc/manual/release-notes/rl-2205.section.md:88]: This won't be backported to a EOL release 22.05, so let's drop this change.

Comment (JohnRTitor) [reply nixos/doc/manual/release-notes/rl-2205.section.md]: Ok looks like manual fails to be built without this

**Comment by adamcstephens** [comment]
Will need another rebase.

**Comment by woffs** [comment]
> Will need another rebase.

done

**Comment by woffs** [comment]
@GrahamcOfBorg test frr

**Comment by woffs** [comment]
@GrahamcOfBorg test frr

**Comment by woffs** [comment]
rebased and squashed

**Comment by woffs** [comment]
rebased and moved the rl changes to a more merge-friendly place in the list

**Comment by woffs** [comment]
I'm really tired from rebasing this again and again

**Comment by woffs** [comment]
- rebased / resolved conflict with 191b68cd26542cfebd561d678ca06b7958c977db / removed "with lib"
- tested with frr-10.0.1 and frr-10.1 (#342259)

**Comment by covert8** [comment]
@Anillc Hi, not to be rude but it would seem that this pr would just need a merge?

**Review by thillux — APPROVED** [review]

**Comment by Anillc** [comment]
> @Anillc Hi, not to be rude but it would seem that this pr would just need a merge?

LGTM. But I don't have permissions to merge this pull request. You may want to ask the committers?

**Comment by woffs** [comment]
Rebased to master because of conflict in the release notes. I do not know if @mweinelt is the one to finally review and merge this PR or if anyone else will be in charge. I guess there are many pull requests around.

**Comment by mweinelt** [comment]
I'm afraid I lack capacity at this time.

**Review by fpletz — APPROVED** [review]

**Comment by woffs** [comment]
Thank you!! 🤸

**Pull request merged**
